style-guide.md

```diff
@@ -0,0 +1,111 @@
+	* If possible, give an alternative for unsupported OSs
+	* "The Dockstore CLI is only supported on Unix-like operating systems, including Linux and Mac. Windows users may wish to..."
+* Do not assume users are or are not using an HPC
+	* Assume that if they are using an HPC, they at least know how to use their HPC's job submission system (SLURM, etc)
+	* If working with/installing something requiring root permissions such as Docker, mention that root permission is needed once, then assume from that point forward that your HPC users have root privileges
+	* However, do not assume that HPC users know how programs you introduce may work differently on an HPC
+		* Example: Running a WDL on an HPC via Cromwell (e.g. via the Dockstore CLI) requires additional runtime arguments that a local user does not need to include in their WDL program, so make sure to either include those runtime arguments in your examples or tell HPC users what additional things they need
+* Assume Linux users are using Ubuntu or know how to translate into their own distro
+	* Exception: If directing very new users to install software via apt-get, consider mentioning that the instructions are specific to Ubuntu -- UCSC's linux lab for students runs on CentOS (which uses yum instead) so this is not a rare occurrence
+* Keep in mind that some of our users are using Macs, so make a note about these situations in the text if they occur:
+	* If there is an important functional difference between the Linux and the Mac version, such as xargs
+	* If there is an important installation difference between the Linux and the Mac version, such as the installation of the Dockstore CLI
+	* If something does not work/works differently on Apple Silicon Macs
+	* If there is an important difference between the bash and zsh version of something
+		* Do not assume that a Mac user is aware of whether they are using bash or zsh (the default changed in Catalina) or knows how to change between the two
+		* Do not worry about other shells
+
+## Coding in RST
+* Write in RST, not markdown
+* Use `.. comment` to write comments
+* Use consistent indentation spacing within a single RST file
+* Do not break up body text with a newline every x characters
+* Use \`this style \<https://example.com\>_` of formatting external hyperlinks, as it is less error prone than the method involving setting the target in another paragraph
+* Use headings in this way throughout the entire repository to avoid issues with embedding RST files into other RST files:
+
+> `page title`  
+> `==========`
+> 
+> `heading level 1`  
+> `---------------`
+>
+> `heading level 2`  
+> `~~~~~~~~~~~~~~~`
+>
+> `heading level 3`  
+> `***************`
+>
+> `heading level 4`  
+> `+++++++++++++++`
+>
+> `heading level 5`  
+> `\``````````````
+>
+ 
+## International
+See also: https://developers.google.com/style/translation 
```

Comment:
there are more links below, do these need to be link-ified?

Reply:
I was going to leave them as URLs but on second thoughts, making them clickable is probably a better idea. Adjusted in https://github.com/dockstore/dockstore-documentation/pull/197/commits/77f141cd066b23caecf30e1a7e14a0a6e303e5a8